### PR TITLE
Minor fixes before release

### PR DESCRIPTION
**Pull request by Ghabry**
~~Don't merge yet,~~ just some minor things that don't hurt to add.

### CODE REVIEW

File: src/game_map.cpp [hunk 1]
```diff
@@ -193,7 +193,7 @@ void Game_Map::Setup(int _id) {
 			break;
 		}
 		if (parent_index < 0) {
-			Output::Warning("Map %d has invalid parent id %d!", Data::treemap.maps[current_index].parent_map);
+			Output::Warning("Map %d has invalid parent id %d!", current_index, Data::treemap.maps[current_index].parent_map);
```
Comment (Ghabry) [inline src/game_map.cpp]: @fmatthew5876 
I'm not 100% sure if the variable I added here is the correct one.

**Comment by Ghabry** [comment]
Thats now ready for review but I still need to check if the ExFont log message is really fixed on Android.

File: src/game_map.cpp (hunk 1 above)
Comment (fmatthew5876) [reply src/game_map.cpp]: You want the id of the map, not the index since maps can have gaps.

`Data::treemap.maps[current_index].ID`

Comment (Ghabry) [reply src/game_map.cpp]: thx, changed it

**Comment by Ghabry** [comment]
found a way to trigger a assert in my RTP code, also fixing this

**Comment by Ghabry** [comment]
somehow logging is unreliable on my Android, sometimes the file isn't written o.O
But I was able to test it, works as expected now. Also tested RTP detection.

**Comment by Ghabry** [comment]
also fixed the UBSan stuff now, nothing big there.

**Comment by fmatthew5876** [comment]
All looks good to me. Lets merge this sucker and get the party started!

**Pull request merged**